### PR TITLE
Add support for Microsoft SQL Server

### PR DESCRIPTION
**Pull request by danrzeppa**
It's been a while in the making, but this pull request has support for almost all of the functionality in node-sql. It took me a while to get around to getting it to a point where it is pushable. I've been using this code in some production systems for a while, however, so I'm confident that it works fairly well.

There have been some additions to the node-sql library, such as support for CASE which I have not implemented yet, but will be doing in the near future.

This is my first pull request, so forgive me if I've done something incorrectly. Since the branch I originally pulled is somewhat old, I'm not sure how well this will merge in. I'm not sure what the best way to handle it is. Perhaps I should merge all the current changes into my branch and then pull request on that?

### CODE REVIEW

**Comment by brianc** [comment]
### OH MY GOD THIS IS AMAZING! :star: :star2: :dizzy: :dancers:

File: lib/node/orderByValue.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ var OrderByColumn = Node.define({
     Node.call(this);
     this.value = config.value;
     this.direction = config.direction;
+		// used when processing OFFSET and LIMIT clauses in MSSQL
+		this.msSQLOffsetNode=undefined;
+		this.msSQLLimitNode=undefined;
```
Comment (brianc) [inline lib/node/orderByValue.js]: little bit of crazy indentation here

File: test/index-tests.js [hunk 2]
```diff
@@ -82,6 +92,7 @@ suite('index', function() {
     var sqliteQuery = mysql.select(user.id).from(user).where(user.email.equals('brian.m.carlson@gmail.com')).toQuery('sqlite');
     var postgresQuery = sqlite.select(user.id).from(user).where(user.email.equals('brian.m.carlson@gmail.com')).toQuery('postgres');
     var mysqlQuery = postgres.select(user.id).from(user).where(user.email.equals('brian.m.carlson@gmail.com')).toQuery('mysql');
+		var mssqlQuery = mysql.select(user.id).from(user).where(user.email.equals('brian.m.carlson@gmail.com')).toQuery('mssql');
```
Comment (brianc) [inline test/index-tests.js]: this line and a few lines above it have really far over tabbed lines that don't match the rest of the indentation.

File: lib/dialect/postgres.js [hunk 3]
```diff
@@ -400,15 +403,27 @@ Postgres.prototype.visitQuery = function(queryNode) {
...
+ */
+Postgres.prototype.visitQueryHelper=function(actions,targets,filters){
+	// lazy-man sorting
+	var sortedNodes = actions.concat(targets).concat(filters);
+	for(var i = 0; i < sortedNodes.length; i++) {
+		var res = this.visit(sortedNodes[i]);
+		this.output = this.output.concat(res);
+	}
+	// implicit 'from'
+	return this.output;
```
Comment (brianc) [inline lib/dialect/postgres.js]: kinda wide spacing here

**Comment by brianc** [comment]
Hey @danrzeppa - this is epic and awesome, truly.  There are a few white space related issues I'm happy to clean up if you don't want to clean up. I'd only suggest you clean them up because then it will make your merge commit cleaner, but again, my editor will auto-format the entire file to 2 spaces no problemo.  I don't want to nitpick on something so small after so much awesome hard work has been done.  Lemme know how you feel about that.

The best and what you get 3,000,000 bonus points for is all the test coverage! That's truly awesome.  I'm very happy to give you contributor access to the repo if you'd like as wel, hopefully to make it easier for to contribute and collaborate in the future.  I'll also edit the README file to reflect it now has support for mssql!

3 cheers!

**Comment by danrzeppa** [comment]
@brianc Sorry about the spacing issues. Some of the other projects I work on use tabs, so usually I set my editor use tabs. I believe I have fixed all of them.

Contributor access would be cool. Initially I would still like another set of eyes on things, before I commit anything permanent; but I do think I have a pretty decent handle on things.

As I mentioned before I'm already looking to update the Microsoft SQL Server support to include the CASE functionality.

**Comment by brianc** [comment]
Yah - we try to always work through pull requests. The only thing I commit directly to master are merge commits, doc changes, & npm version bumps, so you'll still get code reviewed. :smile: 

Pushing this epicness as a new minor version, adding you as contributor, and updating readme now.

**Pull request merged**